code/datums/gamemode/dynamic/dynamic_rulesets.dm

```diff
@@ -116,13 +116,22 @@
 	return FALSE
 
 /datum/dynamic_ruleset/proc/get_weight()
+	var/weight = src.weight
 	weight *= weight_time_day()
-	if(repeatable && weight > 1)
+	if(repeatable)
+		var/halve_weight = FALSE
 		for(var/datum/dynamic_ruleset/DR in mode.executed_rules)
```

Comment:
you could just make a new list that copies the contents of both of these lists and have it loop through that one time, then just do weight /= 2 and eschew the halve_weight var entirely

Reply:
The reason I need two loops is because one is a list of instances of rulesets, the other is a list of ruleset type paths.

Reply:
you could operate on it with a typecheck for marginally improved readability and reduced lines, I think, but it doesn't really matter I suppose